bundle-workflow/tests/test_aws/test_s3_bucket.py

```diff
@@ -0,0 +1,189 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import unittest
+from unittest.mock import MagicMock, call, patch
+
+from botocore.exceptions import ClientError
+
+from aws.s3_bucket import S3Bucket, S3DownloadError, STSError
+
+mock_sts = MagicMock()
+mock_s3_resource = MagicMock()
+mock_s3_client = MagicMock()
+bucket_name = "unitTestBucket"
```

Comment:
Any reason to reuse these globals instead of just doing `MagicMock()` in the side effect code?

Reply:
Yeah, somehow the `mock_s3_resource` wasn't picking the same mock object with MagicMock inside side effect. Will revisit this later and fix.